src/services/workspace-client/devWorkspaceClient.ts

```diff
@@ -149,10 +155,54 @@ export class DevWorkspaceClient extends WorkspaceClient {
           ]
         },
         spec: pluginDevfile
-      });
+      };
+      devWorkspaceTemplates.push(theiaDWT);
+    }
 
-      this.addPlugin(createdWorkspace, pluginName, theiaDWT.metadata.namespace);
+    const devWorkspace: V1alpha2DevWorkspace = createdWorkspace;
+    // call theia library to insert all the logic
+    const inversifyBindings = new InversifyBinding();
+    const container = await inversifyBindings.initBindings({
+      pluginRegistryUrl: pluginRegistryUrl || '',
+      axiosInstance: this.axios,
+      insertTemplates: false,
+    });
+    const cheTheiaPluginsContent = optionalFilesContent['.che/che-theia-plugins.yaml'];
+    const vscodeExtensionsJsonContent = optionalFilesContent['.vscode/extensions.json'];
+    const cheTheiaPluginsDevfileResolver = container.get(CheTheiaPluginsDevfileResolver);
+
+    let sidecarPolicy: SidecarPolicy;
+    const devfileCheTheiaSidecarPolicy = (devfile as V1alpha2DevWorkspaceSpecTemplate).attributes?.['che-theia.eclipse.org/sidecar-policy'];
+    if (devfileCheTheiaSidecarPolicy === 'USE_DEV_CONTAINER') {
+      sidecarPolicy = SidecarPolicy.USE_DEV_CONTAINER;
+    } else {
+      sidecarPolicy = SidecarPolicy.MERGE_IMAGE;
     }
+    console.debug('Loading devfile', devfile, 'with optional .che/che-theia-plugins.yaml', cheTheiaPluginsContent, 'and .vscode/extensions.json', vscodeExtensionsJsonContent, 'with sidecar policy', sidecarPolicy);
+    // call library to update devWorkspace and add optional templates
+    await cheTheiaPluginsDevfileResolver.handle({
+      devfile,
+      cheTheiaPluginsContent,
+      vscodeExtensionsJsonContent,
+      devWorkspace,
+      devWorkspaceTemplates,
+      sidecarPolicy,
+      suffix: workspaceId,
+    });
+
+    await Promise.all(devWorkspaceTemplates.map(async template => {
+      // if no namespace, update it
```

Comment:
Should we really try to create dwt in the namespace different than devworkspace one?

Reply:
probably not

Reply:
then let's remote if statement and always override namespace.

Reply:
fixed